chapter04/chapter04.adoc

```diff
@@ -0,0 +1,317 @@
+
+When we access the `http://localhost:8080/openapi` URL, we should see the API documentation that was generated by MicroProfile OpenAPI:
+
+[source, yaml]
+----
+openapi: 3.0.3
+info:
+  title: Generated API
+  version: "1.0"
+servers:
+- url: http://localhost:5050/mp-ecomm-store
+paths:
+  /api/products:
+    get:
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Product'
+    put:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Product'
+      responses:
+        "200":
+          description: OK
+    post:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Product'
+      responses:
+        "200":
+          description: OK
+  /api/products/products/{id}:
+    delete:
+      parameters:
+      - name: id
+        in: path
+        required: true
+        schema:
+          format: int64
+          type: integer
+      responses:
+        "200":
+          description: OK
+  /api/products/{id}:
+    get:
+      parameters:
+      - name: id
+        in: path
+        required: true
+        schema:
+          format: int64
+          type: integer
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Product'
+components:
+  schemas:
+    Product:
+      required:
+      - name
+      - description
+      - price
+      type: object
+      properties:
+        id:
+          format: int64
+          type: integer
+        name:
+          type: string
+        description:
+          type: string
+        price:
+          format: double
+          type: number
+----
+
+As we can see, MicroProfile OpenAPI has generated API documentation for our resource class. We can use this documentation to learn about the API and how to use it. 
+
+MicroProfile OpenAPI allows developers to produce these specifications directly from their codebase, leveraging annotations and/or providing OpenAPI documents statically. This direct generation ensures that the API documentation is always up to date with the code.
+
+=== Exploring the APIs using Swagger UI
+
+To open Swagger UI for the API documentation generated using MicroProfile OpenAPI, you will need to deploy your application to a server that supports MicroProfile, such as Open Liberty, WildFly, Quarkus, or Payara Micro. These servers automatically generate the OpenAPI documentation for your RESTful services based on the annotations in your code. 
+
+Next, visit the following URL to launch the Swagger UI: 
+
+----
+http://localhost:8080/openapi/ui
```

Comment:
In Open Liberty, the port number is 9080. You might want to use <port> instead.

Reply:
Fixed with [c2128ed](https://github.com/ttelang/microprofile-tutorial/commit/c2128ed294e0008a58a54142aa74181607b029d6)